link error: undefined reference to `boost::gregorian::greg_month::get_month_map_ptr()

can not compile i2pd on a relatively new/up-to-date install of Gentoo.
the emerge fails with:

Linking CXX executable i2pd
/usr/bin/cmake -E cmake_link_script CMakeFiles/i2pd-bin.dir/link.txt --verbose=1
/usr/bin/x86_64-pc-linux-gnu-g++   -march=native -O2 -pipe  -Wformat -Wformat-security -Werror=format-security -fstack-protector --param ssp-buffer-size=4   -Wl,-O1 -Wl,--as-needed -z relro -z now CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/AddressBook.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/CryptoConst.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Garlic.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/I2NPProtocol.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Identity.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/LeaseSet.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Log.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/NTCPSession.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/NetDb.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Profiling.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Reseed.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/RouterContext.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/RouterInfo.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/SSU.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/SSUData.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/SSUSession.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Streaming.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Destination.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/TransitTunnel.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Tunnel.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/TunnelGateway.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Transports.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/TunnelEndpoint.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/TunnelPool.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/aes.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/base64.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/util.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Datagram.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/BOB.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/ClientContext.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Daemon.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/HTTPProxy.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/HTTPServer.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/I2PService.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/I2PControl.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/I2PTunnel.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/SAM.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/SOCKS.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/UPnP.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/i2p.cpp.o CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/DaemonLinux.cpp.o  -o i2pd -rdynamic -lboost_system-mt -lboost_filesystem-mt -lboost_regex-mt -lboost_program_options-mt -lcryptopp -lpthread 
CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Profiling.cpp.o: In function `boost::date_time::ymd_formatter<boost::date_time::year_month_day_base<boost::gregorian::greg_year, boost::gregorian::greg_month, boost::gregorian::greg_day>, boost::date_time::simple_format<char>, char>::ymd_to_string(boost::date_time::year_month_day_base<boost::gregorian::greg_year, boost::gregorian::greg_month, boost::gregorian::greg_day>)':
Profiling.cpp:(.text._ZN5boost9date_time13ymd_formatterINS0_19year_month_day_baseINS_9gregorian9greg_yearENS3_10greg_monthENS3_8greg_dayEEENS0_13simple_formatIcEEcE13ymd_to_stringES7_[_ZN5boost9date_time13ymd_formatterINS0_19year_month_day_baseINS_9gregorian9greg_yearENS3_10greg_monthENS3_8greg_dayEEENS0_13simple_formatIcEEcE13ymd_to_stringES7_]+0xc1): undefined reference to `boost::gregorian::greg_month::as_short_string() const'
CMakeFiles/i2pd-bin.dir/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999/Profiling.cpp.o: In function `unsigned short boost::date_time::month_str_to_ushort<boost::gregorian::greg_month>(std::string const&)':
Profiling.cpp:(.text._ZN5boost9date_time19month_str_to_ushortINS_9gregorian10greg_monthEEEtRKSs[_ZN5boost9date_time19month_str_to_ushortINS_9gregorian10greg_monthEEEtRKSs]+0x12f): undefined reference to `boost::gregorian::greg_month::get_month_map_ptr()'
collect2: error: ld returned 1 exit status
CMakeFiles/i2pd-bin.dir/build.make:1093: recipe for target 'i2pd' failed
make[2]: *** [i2pd] Error 1
make[2]: Leaving directory '/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999-abi_x86_64.amd64'
CMakeFiles/Makefile2:63: recipe for target 'CMakeFiles/i2pd-bin.dir/all' failed
make[1]: *** [CMakeFiles/i2pd-bin.dir/all] Error 2
make[1]: Leaving directory '/var/tmp/portage/net-misc/i2pd-9999/work/i2pd-9999-abi_x86_64.amd64'
Makefile:116: recipe for target 'all' failed


-sokolowskim

--- a/build/CMakeLists.txt
+++ b/build/CMakeLists.txt
@@ -124,7 +124,7 @@ endif()
 # libraries
 find_package ( Threads REQUIRED )
 
-find_package ( Boost COMPONENTS system filesystem regex program_options REQUIRED )
+find_package ( Boost COMPONENTS system filesystem regex program_options date_time REQUIRED )
 if(NOT DEFINED Boost_INCLUDE_DIRS)
   message(SEND_ERROR "Boost is not found, or your boost version was bellow 1.46. Please download Boost!")
 endif()

